packages/google_maps_flutter/google_maps_flutter_web/lib/src/heatmaps.dart

```diff
@@ -0,0 +1,63 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of google_maps_flutter_web;
+
+/// This class manages all the [HeatmapController]s associated to a [GoogleMapController].
+class HeatmapsController extends GeometryController {
+  // A cache of [HeatmapController]s indexed by their [HeatmapId].
+  final Map<HeatmapId, HeatmapController> _heatmapIdToController;
+
+  /// Initialize the cache
+  HeatmapsController()
+      : _heatmapIdToController = Map<HeatmapId, HeatmapController>();
+
+  /// Returns the cache of [HeatmapController]s. Test only.
+  @visibleForTesting
+  Map<HeatmapId, HeatmapController> get heatmaps => _heatmapIdToController;
+
+  /// Adds a set of [Heatmap] objects to the cache.
+  ///
+  /// Wraps each [Heatmap] into its corresponding [HeatmapController].
+  void addHeatmaps(Set<Heatmap> heatmapsToAdd) {
+    heatmapsToAdd.forEach((heatmap) {
+      _addHeatmap(heatmap);
+    });
+  }
+
+  void _addHeatmap(Heatmap heatmap) {
+    if (heatmap == null) {
+      return;
+    }
+
+    final populationOptions = _heatmapOptionsFromHeatmap(heatmap);
```

Comment:
I've renamed all these, populationOptions doesn't make too much sense to me:

```suggestion
    final heatmapOptions = _heatmapOptionsFromHeatmap(heatmap);
```

Reply:
Done